### PR TITLE
Lab2

### PR DESCRIPTION
**Pull request by danmory**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+venv
+__pycache__
```
Comment (bkmceh) [inline .gitignore]: Are you sure that leave .gitignore file visible is a good idea?

File: app_go/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+COPY *.go ./
+
+RUN go build -o /go_server
+
+EXPOSE 8080
+
+ENV GIN_MODE release
+
+CMD ["/go_server"]
```
Comment (bkmceh) [inline app_go/Dockerfile]: No newline at the end of file

File: .gitignore (hunk 1 above)
Comment (bkmceh) [reply .gitignore]: Also no newline at the end of file

File: app_go/Dockerfile.multistage [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+COPY --from=build /go_server /go_server
+
+EXPOSE 8080
+
+USER nonroot:nonroot
+
+ENV GIN_MODE release
+
+ENTRYPOINT ["/go_server"]
```
Comment (bkmceh) [inline app_go/Dockerfile.multistage]: Same, no newline
Check this: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: app_python/.dockerignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+PYTHON.md
+__pycache__
+DOCKER.md
```
Comment (bkmceh) [inline app_python/.dockerignore]: Same, no newline

File: app_python/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+WORKDIR /app
+
+COPY requirements.txt requirements.txt
+RUN pip3 install --no-cache-dir --upgrade -r requirements.txt
+
+COPY . .
+
+HEALTHCHECK --interval=10s --timeout=10s --retries=2 CMD curl -sS 127.0.0.1:8000 || exit 1
+
+CMD [ "uvicorn", "main:app", "--host", "0.0.0.0"]
```
Comment (bkmceh) [inline app_python/Dockerfile]: Same, no newline

File: .gitignore (hunk 1 above)
Comment (danmory) [reply .gitignore]: https://stackoverflow.com/questions/5765645/should-you-commit-gitignore-into-the-git-repos check this

**Comment by ezio-42** [comment]
LGTM

**Pull request closed without merging**